search/searcher/search_disjunction.go

```diff
@@ -46,6 +46,18 @@ func optionsDisjunctionOptimizable(options search.SearcherOptions) bool {
 func newDisjunctionSearcher(ctx context.Context, indexReader index.IndexReader,
 	qsearchers []search.Searcher, min float64, options search.SearcherOptions,
 	limit bool) (search.Searcher, error) {
+
+	// The KNN Searcher optimization is a necessary pre-req for KNN Searchers, not an
+	// optional optimization like for, say term searchers.
+	// It's an optimization in the sense that it's more optimal to batch calls
+	// Hence, this optimization should be invoked for KNN searchers irrespective
+	// of whether there are other searchers in the disjunction or not.
+
+	err := optimizeKNNSearcher(ctx, "", indexReader, qsearchers, options)
```

Comment:
TODO Need to double check nested queries of the type, to avoid double optimization:
{
  disjunction: {
     disjunct1: disjunction: {term,KNN}
     disjunct2: KNN
 }
}